From owner-linux-pci@atrey.karlin.mff.cuni.cz Sun Dec 11 16:15:28 2005
Date: Mon, 12 Dec 2005 11:12:01 +1100
From: Anton Blanchard <anton@samba.org>
To: Greg Kroah-Hartman <gregkh@suse.de>
Cc: <miltonm@bga.com>, <rajesh.shah@intel.com>
Subject: PCI express must be initialized before PCI hotplug
Message-ID: <20051212001201.GF23641@krispykreme>
Content-Disposition: inline


From: Milton Miller <miltonm@bga.com>

PCI express hotplug uses the pcieportbus driver so pcie must be
initialized before hotplug/.  This patch changes the link order.

Signed-Off-By: Milton Miller <miltonm@bga.com>
Acked-by: Anton Blanchard <anton@samba.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/pci/Makefile |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

--- gregkh-2.6.orig/drivers/pci/Makefile
+++ gregkh-2.6/drivers/pci/Makefile
@@ -6,6 +6,9 @@ obj-y		+= access.o bus.o probe.o remove.
 			pci-driver.o search.o pci-sysfs.o rom.o setup-res.o
 obj-$(CONFIG_PROC_FS) += proc.o
 
+# Build PCI Express stuff if needed
+obj-$(CONFIG_PCIEPORTBUS) += pcie/
+
 obj-$(CONFIG_HOTPLUG) += hotplug.o
 
 # Build the PCI Hotplug drivers if we were asked to
@@ -40,7 +43,3 @@ endif
 ifeq ($(CONFIG_PCI_DEBUG),y)
 EXTRA_CFLAGS += -DDEBUG
 endif
-
-# Build PCI Express stuff if needed
-obj-$(CONFIG_PCIEPORTBUS) += pcie/
-
